pkg/log/zap/flags.go

```diff
@@ -158,3 +166,34 @@ func (v sampleValue) IsBoolFlag() bool {
 func (v sampleValue) Type() string {
 	return "sample"
 }
+
+type timeformatValue struct {
+	set bool
+	str string
```

Comment:
We can just go ahead and set the `zapcore.TimeEncoder` in the `Set` function, right? So this struct could be:

```go
type timeformatValue struct {
	set bool
	timeEncoder zapcore.TimeEncoder
	str string
}
```

I think the `timeformatValue.Set` function can end up looking a lot like `encoderValue.Set`

Reply:
That's a really great suggestion - much more elegant! I've been able to get that working and now you can pass --zap-timeformat=iso8601,ISO8601,millis,nanos or if you set nothing (or something random) it defaults to 'epoch'.